packages/typescript/src/index.js

```diff
@@ -58,6 +65,66 @@ module.exports = {
       },
     ],
 
+    // Recommended rules that require type information
+    '@typescript-eslint/no-unsafe-argument': 'off',
+    '@typescript-eslint/no-unsafe-assignment': 'off',
+    '@typescript-eslint/no-unsafe-call': 'off',
+    '@typescript-eslint/no-unsafe-member-access': 'off',
+    '@typescript-eslint/no-unsafe-return': 'off',
+    '@typescript-eslint/restrict-template-expressions': 'off',
+    '@typescript-eslint/unbound-method': 'off',
```

Comment:
This rule is causing a lot of confusing errors like this:

``` 
  14:36  error  Avoid referencing unbound methods which may cause unintentional scoping of `this`.
If your function does not access `this`, you can annotate it with `this: void`, or consider using an arrow function instead  @typescript-eslint/unbound-method
```

I don't think this is actually an issue, so I have disabled this rule.

Reply:
You may be aware but it's an easy enough mistake to do:
```
class Foo {
  public logThis() {
    console.log(this);
  }
}
const foo = new Foo();

foo.logThis(); // logs foo
window.setTimeout(foo.logThis); // logs global 

window.setTimeout(foo.logThis.bind()); // do this 
window.setTimeout(() => foo.logThis()); // or this 


```

Is it worth silencing individually where it's actually desired?

Reply:
The problem is that it causes errors in situations like this:

```ts
expect(Math.abs).toStrictEqual(rootRealmGlobal.Math.abs);
```

Which we happen to do quite a bit in `snaps-monorepo`. We could just disable it for that repo, but I would guess that there are others with a similar problem.

Reply:
Hmm. Are examples like that widespread? We can disable the rule for specific files/lines. It does seem like what the rule is suggesting is what we are already doing in most places.

Reply:
It seemed to happen quite a bit in `snaps-monorepo`. Not sure about other repos.

Reply:
I guess that this can be revisited if/when Typescript stdlib is annotating consistently to make those functions work.

https://www.typescriptlang.org/docs/handbook/2/functions.html#declaring-this-in-a-function

https://github.com/microsoft/TypeScript/blob/c0f8d1cf75f85132ed46950de884d97ec5312ae8/lib/lib.es5.d.ts#L665